scene/resources/curve.cpp

```diff
@@ -1791,6 +1800,22 @@ real_t Curve3D::_sample_baked_tilt(Interval p_interval) const {
 	return Math::lerp(r[idx], r[idx + 1], frac);
 }
 
+// Internal method for getting posture at a baked point. Assuming caller
+// make all sanity checks.
+Basis Curve3D::_get_posture_baked(int p_baked_idx) const {
+	Vector3 forward = baked_forward_vector_cache[p_baked_idx];
+
+	Vector3 up;
+	if (up_vector_enabled) {
+		up = baked_up_vector_cache[p_baked_idx];
+	} else {
+		up = Vector3(0.0, 1.0, 0.0);
+	}
+
+	const Basis frame = Basis::looking_at(forward, up);
+	return frame;
+}
+
```

Comment:
This method looks like it could be implemented as a case by argument of `_sample_posture()`. ex) `Basis Curve3D::_sample_posture(Interval p_interval, bool p_apply_tilt, bool p_simplify = false)`

Reply:
`_get_posture_baked` is not a special case of `_sample_posture`, rather a dependency for implementing `_sample_posture`. See line 1830, 1831. It is factored out to reduce code repeatition.

Reply:
Oh sure. But the naming is obviously strange. A better name would be something like `_compose_posture`.

It is fine to remove the underscore like `compose_posture()` and put it in public so that EditorPlugin can use it, but you can't bind the method.

Reply:
> A better name would be something like _compose_posture.

Ok

> It is fine to remove the underscore like compose_posture() and put it in public so that EditorPlugin can use it (as said above, points_in_cache and get_point_baked_posture() should be implemented on the EditorPlugin) as long as it is not bound as a gdscript method.

we can expose `compose_posture` for the `EditorPlugin`. But the `EditorPlugin` still need to know the correspondence  between baked points and control points. These information is known only in the curve baking process(`Curve3D::_bake()`) and currently cached in `Curve3D::points_in_cache`. 

If we move `points_in_cache` and `get_point_baked_posture()` to EditorPlugin, how do we communicate this information from Curve3D to EditorPlugin?

Reply:
since the main difference of opinions is discussed in other thread. I close this for readability.